packages/edit-site/src/components/page-templates/dataviews-templates.js

```diff
@@ -0,0 +1,207 @@
+			per_page: -1,
+		} );
+	const { shownTemplates, paginationInfo } = useMemo( () => {
+		if ( ! allTemplates )
+			return {
+				shownTemplates: EMPTY_ARRAY,
+				paginationInfo: { totalItems: 0, totalPages: 0 },
+			};
+		let filteredTemplates = [ ...allTemplates ];
+		// Handle global search.
+		if ( view.search ) {
+			const normalizedSearch = normalizeSearchInput( view.search );
+			filteredTemplates = filteredTemplates.filter( ( item ) => {
+				const title = item.title?.rendered || item.slug;
+				return (
+					normalizeSearchInput( title ).includes(
+						normalizedSearch
+					) ||
+					normalizeSearchInput( item.description ).includes(
+						normalizedSearch
+					)
+				);
+			} );
+		}
+		// Handle sorting.
+		// TODO: Explore how this can be more dynamic..
+		if ( view.sort ) {
+			if ( view.sort.field === 'title' ) {
+				filteredTemplates.sort( ( a, b ) => {
+					const titleA = a.title?.rendered || a.slug;
+					const titleB = b.title?.rendered || b.slug;
+					return view.sort.direction === 'asc'
+						? titleA.localeCompare( titleB )
+						: titleB.localeCompare( titleA );
+				} );
+			}
+		}
+		// Handle pagination.
+		const start = ( view.page - 1 ) * view.perPage;
+		const totalItems = filteredTemplates?.length || 0;
+		filteredTemplates = filteredTemplates?.slice(
+			start,
+			start + view.perPage
+		);
+		return {
+			shownTemplates: filteredTemplates,
+			paginationInfo: {
+				totalItems,
+				totalPages: Math.ceil( totalItems / view.perPage ),
+			},
+		};
+	}, [ allTemplates, view ] );
+	const fields = useMemo(
+		() => [
+			{
+				header: __( 'Template' ),
+				id: 'title',
+				getValue: ( { item } ) => item.title?.rendered || item.slug,
+				render: ( { item } ) => {
+					return (
+						<VStack spacing={ 1 }>
+							<Heading as="h3" level={ 5 }>
+								<Link
+									params={ {
+										postId: item.id,
+										postType: item.type,
+										canvas: 'edit',
+									} }
+								>
+									{ decodeEntities(
+										item.title?.rendered || item.slug
+									) || __( '(no title)' ) }
+								</Link>
+							</Heading>
+						</VStack>
+					);
+				},
+				maxWidth: 400,
+				sortingFn: 'alphanumeric',
+				enableHiding: false,
+			},
+			{
+				header: __( 'Description' ),
+				id: 'description',
+				getValue: ( { item } ) => item.description,
+				render: ( { item } ) => {
+					return (
+						item.description && (
+							<Text variant="muted">
+								{ decodeEntities( item.description ) }
+							</Text>
+						)
+					);
+				},
+				enableSorting: false,
+			},
+			{
+				header: __( 'Author' ),
+				id: 'author',
+				getValue: () => {},
+				render: ( { item } ) => {
+					return (
+						<AddedBy postType={ item.type } postId={ item.id } />
+					);
+				},
+				enableHiding: false,
+				enableSorting: false,
+			},
+		],
+		[]
+	);
+	const resetTemplateAction = useResetTemplateAction();
+	const trashTemplateAction = useTrashTemplateAction();
```

Comment:
Without the experiment enabled, the remove action requires confirmation, should we do the same? Also, unlike pages, this is not sent to trash but removed permanently, so perhaps `removeTemplateAction` would be a better name?

Reply:
I'm working on the actions right now and will push really soon. You're right about the name. I'll update.

Reply:
Ah, gotcha, I was reviewing them now. I'll wait until you push.